test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp

```diff
@@ -1488,6 +1488,91 @@ TEST_F (OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, PropA
+        const Instant startInstant = Instant::DateTime(DateTime::Parse("2023-01-01 00:00:00.000"), Scale::UTC) ;
+
+        // Reference data setup
+        const Table referenceData = Table::Load(File::Path(Path::Parse("/app/test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/Orekit_Drag_Exponential_2hr_run.csv")), Table::Format::CSV, true) ;
+
+        Array<Instant> instantArray = Array<Instant>::Empty() ;
+        Array<Vector3d> referencePositionArray_GCRF = Array<Vector3d>::Empty() ;
+        Array<Vector3d> referenceVelocityArray_GCRF = Array<Vector3d>::Empty() ;
+
+        for (const auto& referenceRow : referenceData)
+        {
+
+                instantArray.add(Instant::DateTime(DateTime::Parse(referenceRow[0].accessString(), DateTime::Format::ISO8601) , Scale::UTC)) ;
+
+                referencePositionArray_GCRF.add( Vector3d(referenceRow[1].accessReal(), referenceRow[2].accessReal(), referenceRow[3].accessReal()) ) ;
+                referenceVelocityArray_GCRF.add( Vector3d(referenceRow[4].accessReal(), referenceRow[5].accessReal(), referenceRow[6].accessReal()) ) ;
+
+        }
+
+        // Create environment
+        const Instant instantJ2000 = Instant::J2000() ;
+        const Array<Shared<Object>> objects =
+        {
+            std::make_shared<Earth>(Earth::Spherical())
+        } ;
+
+        const Environment customEnvironment = Environment(instantJ2000, objects) ;
+
+        // Setup initial conditions
+        const State state = { startInstant, Position::Meters({ referencePositionArray_GCRF[0] }, gcrfSPtr_), Velocity::MetersPerSecond({ referenceVelocityArray_GCRF[0] }, gcrfSPtr_) } ;
+
+        // Satellite dynamics setup
+        SatelliteDynamics satelliteDynamics = { customEnvironment, satelliteSystem } ;
+
+        // RK4 Numerical solver
+        NumericalSolver numericalSolver = { NumericalSolver::LogType::NoLog, NumericalSolver::StepperType::RungeKutta4, 30.0, 1.0e-15, 1.0e-15 } ;
+
+        // Setup Propagator model and orbit
+        const Propagator propagator = { satelliteDynamics, numericalSolver_ } ;
+
+        // Propagate all states
+        const Array<State> propagatedStateArray = propagator.calculateStatesAt(state, instantArray) ;
+
+        // Validation loop
+        for (size_t i = 0; i < instantArray.getSize(); i++)
+        {
+            // GCRF Compare
+            const Position position_GCRF = propagatedStateArray[i].accessPosition() ;
+            const Velocity velocity_GCRF = propagatedStateArray[i].accessVelocity() ;
+
+            const double positionError_GCRF = (position_GCRF.accessCoordinates() - referencePositionArray_GCRF[i]).norm() ;
+            const double velocityError_GCRF = (velocity_GCRF.accessCoordinates() - referenceVelocityArray_GCRF[i]).norm() ;
+
+            ASSERT_EQ(*Frame::GCRF(), *position_GCRF.accessFrame()) ;
+            ASSERT_EQ(*Frame::GCRF(), *velocity_GCRF.accessFrame()) ;
+
+            ASSERT_GT(2e-4, positionError_GCRF) ;
```

Comment:
Was this the lowest error we could get?
Considering that we have 1e-12 error for the atmospheric model I would expect this to be closed for a 2 body spherical earth. Maybe the GM value is different?

Reply:
Yep this is the tightest tolerance it would take. Everything I can tell (GM included) is matching. I think OREKit is doing something at least a _little_ weird under the hood, because their propagation results in a mild Z-axis offset for this simple 2-force propagation. (hence the 1e-12 tolerance on the force model test, rather than 1e-15). 

This is a comparable (slightly tighter) tolerance to our [test of the spherical Earth/Moon/Sun system](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp#L1416).

This tolerance seems pretty in-line with the accuracy we expect in the other validation tests IMO. Possible that something is off somewhere, though.

Reply:
Yeah agree. Ideally we'd be pretty lock-step. Maybe I'll try adding a dummy simple Spherical Earth vs. Spherical Earth as a sanity check.